### PR TITLE
[YUNIKORN-2162] Add unit test for checkHealthStatus

### PR DESCRIPTION
**Pull request by brandboat**
### What is this PR for?
- Add unit test for `checkHealthStatus`
- Remove unused method in scheduler health
- Rename `scheduler-health.go` to `scheduler_health.go`

### What type of PR is it?
* [x] - Improvement

### Todos
N/A

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2162

### How should this be tested?
covered by unit test

### Screenshots (if appropriate)
N/A

### Questions:
N/A

### CODE REVIEW

File: pkg/webservice/handlers_test.go [hunk 1]
```diff
@@ -1720,3 +1720,83 @@ func verifyStateDumpJSON(t *testing.T, aggregated *AggregatedStateInfo) {
...
+	assert.NilError(t, err, unmarshalError)
+	assert.Equal(t, http.StatusNotFound, errInfo.StatusCode, statusCodeError)
+	assert.Equal(t, "The healthy status of scheduler is not found", errInfo.Message, jsonMessageError)
+}
+
+func TestCheckHealthStatus(t *testing.T) {
+	testCases := []struct {
+		name     string
+		expected *dao.SchedulerHealthDAOInfo
+	}{
```
Comment (pbacsko) [inline pkg/webservice/handlers_test.go:1742]: Don't think we need this overhead for only two test cases. Just call `runHealthChecks()` directly twice.

File: pkg/webservice/handlers_test.go [hunk 2]
```diff
@@ -1720,3 +1720,83 @@ func verifyStateDumpJSON(t *testing.T, aggregated *AggregatedStateInfo) {
...
+	req, err := http.NewRequest("GET", "/ws/v1/scheduler/healthcheck", strings.NewReader(""))
+	assert.NilError(t, err, "Error while creating the healthcheck request")
+	resp := &MockResponseWriter{}
+	checkHealthStatus(resp, req)
+
+	var errInfo dao.YAPIError
+	err = json.Unmarshal(resp.outputBytes, &errInfo)
+	assert.NilError(t, err, unmarshalError)
+	assert.Equal(t, http.StatusNotFound, errInfo.StatusCode, statusCodeError)
+	assert.Equal(t, "The healthy status of scheduler is not found", errInfo.Message, jsonMessageError)
```
Comment (pbacsko) [inline pkg/webservice/handlers_test.go:1735]: Could you change this text to sth else? It sounds funny.
I propose "Health check is not available".

**Review by pbacsko — CHANGES_REQUESTED** [review]
Two things.

File: pkg/webservice/handlers_test.go (hunk 2 above)
Comment (brandboat) [reply pkg/webservice/handlers_test.go]: Already addressed comments in the latest commit, many thanks !

**Review by pbacsko — COMMENTED** [review]
+1

**Pull request closed without merging**